libs/language-server/src/lib/ast/wrappers/pipeline-wrapper.ts

```diff
@@ -0,0 +1,141 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { strict as assert } from 'assert';
+
+import {
+  BlockDefinition,
+  CompositeBlocktypeDefinition,
+  PipelineDefinition,
+} from '../generated/ast';
+
+import { AstNodeWrapper } from './ast-node-wrapper';
+import { PipeWrapper, createWrappersFromPipeChain } from './pipe-wrapper';
+
+export class PipelineWrapper
+  implements AstNodeWrapper<PipelineDefinition | CompositeBlocktypeDefinition>
+{
+  public readonly astNode: PipelineDefinition | CompositeBlocktypeDefinition;
```

Comment:
I thought about transforming it into the following so that astNode is typed better. I'm not sure if we have a benefit from that though.

```suggestion
export class PipelineWrapper<
  T extends PipelineDefinition | CompositeBlocktypeDefinition,
> implements AstNodeWrapper<T>
{
  public readonly astNode: T;
```

Reply:
Awesome, seems to work now after removing the `SimplePipe` syntax in #487